### PR TITLE
Better cross sheet handling and ability to handle open ended ranges.

### PR DESCRIPTION
**Pull request by jordanvance**
Calling `set_sheet` in the evaluator caused evaluation to fail when there was a lot of back and forth. Better to pass in the empty context and let the evaluator figure it out. `cross_sheet.xlsx` now has tests to cover this.

Also allow for open ended ranges (e.g. A:A) which previously would be ignored by evaluation. Because this can be *incredibly* expensive, we err on the side of dense spreadsheets and cut off evaluation after we hit 1000 empty rows or columns.

### CODE REVIEW

File: xlcalculator/utils.py [hunk 1]
```diff
@@ -47,7 +47,16 @@ def resolve_ranges(ranges, default_sheet='Sheet1'):
                 min_col is None or min_row is None
                 or max_col is None or max_row is None
         ):
```
Comment (strichter) [inline xlcalculator/utils.py:49]: This overreaching can now be removed.

File: xlcalculator/ast_nodes.py [hunk 2]
```diff
@@ -146,13 +148,23 @@ def eval(self, context):
...
+                    if cell.value == '' or cell.value is None:
+                        empty_col += 1
+                        if empty_col > MAX_EMPTY:
+                            break
+                    row_cells.append(cell)
+                if not row_cells:
+                    empty_row += 1
+                    if empty_row > MAX_EMPTY:
+                        break
+                range_cells.append(row_cells)
```
Comment (strichter) [inline xlcalculator/ast_nodes.py:167]: This maxes at total empty rows/cols not empty rows/cols after last non-empty row/cols.

Comment (strichter) [inline xlcalculator/ast_nodes.py:167]: This maxes at total empty rows/cols not empty rows/cols after last non-empty row/cols.

File: xlcalculator/xlfunctions/func_xltypes.py [hunk 3]
```diff
@@ -279,7 +279,7 @@ def __le__(self, other):
 
     def __eq__(self, other):
         if self.value in (None, '') \
-                and (other in (None, '') or isinstance(other, None)):
+                and (other in (None, '') or other is None):
```
Comment (strichter) [inline xlcalculator/xlfunctions/func_xltypes.py:282]: This simply duplicates in the "contains" expression before thus making it superfluous.

File: xlcalculator/ast_nodes.py (hunk 2 above)
Comment (jordanvance) [reply xlcalculator/ast_nodes.py:167]: ah right.

File: xlcalculator/ast_nodes.py [hunk 4]
```diff
@@ -146,13 +148,27 @@ def eval(self, context):
...
+                            break
+                    else:
+                        empty_col = 0
+                    row_cells.append(cell)
+                if not row_cells:
+                    empty_row += 1
+                    if empty_row > MAX_EMPTY:
+                        break
+                    else:
+                        empty_col = 0
```
Comment (strichter) [inline xlcalculator/ast_nodes.py:170]: Why this?

**Comment by strichter** [comment]
Looks good to me otherwise. I have done a another review of `set_sheet()` and its use and it looks good to me.

File: xlcalculator/ast_nodes.py (hunk 4 above)
Comment (jordanvance) [reply xlcalculator/ast_nodes.py]: because I was rushing. oops!

**Pull request merged**